src/search.js

```diff
@@ -322,43 +473,61 @@ class Search {
         }
         else if (backwards) {
             var forEachInLine = function(row, endIndex, callback) {
-                var line = session.getLine(row);
-                var matches = [];
-                var m, last = 0;
-                re.lastIndex = 0;
-                while((m = re.exec(line))) {
-                    var length = m[0].length;
-                    last = m.index;
-                    if (!length) {
-                        if (last >= line.length) break;
-                        re.lastIndex = last += lang.skipEmptyMatch(line, last, supportsUnicodeFlag);
-                    }
-                    if (m.index + length > endIndex)
-                        break;
-                    matches.push(m.index, length);
-                }
-                for (var i = matches.length - 1; i >= 0; i -= 2) {
-                    var column = matches[i - 1];
-                    var length = matches[i];
-                    if (callback(row, column, row, column + length))
+                if (multiline) {
```

Comment:
should we reset regex lastIndex above this line just to reset it?
 re.lastIndex = 0;

Reply:
Yes, see: https://stackoverflow.com/a/4724920/7598333

Reply:
Yes, see: https://stackoverflow.com/a/4724920

Reply:
So can we add
re.lastIndex = 0;

Reply:
Sorry, I wasn’t focused earlier.

No need, because it’s already set in `multiLineBackwardMatch`.